pytest_html/plugin.py

```diff
@@ -320,13 +331,19 @@ def generate_summary_item(self):
             unicode_doc = unicode_doc.decode('utf-8')
         return unicode_doc
 
-    def _save_report(self, report_content):
-        if not os.path.exists(os.path.dirname(self.logfile)):
-            os.makedirs(os.path.dirname(self.logfile))
-        logfile = open(self.logfile, 'w', encoding='utf-8')
+    def _save_report(self, report_content, self_contained_html):
+        dir_name = os.path.dirname(self.logfile)
+        if not os.path.exists(dir_name):
+            os.makedirs(dir_name)
+        with open(self.logfile, 'w', encoding='utf-8') as f:
+            f.write(report_content)
+        f.close()
```

Comment:
When using "with ... as" in python, you don't need to close it afterwards, that is done automatically.

```python
>>> with open('test.txt', 'w') as f:
...     f.write('foo')
... 
3
>>> f.closed
True
```